Mathlib/Topology/Category/Stonean/Basic.lean

```diff
@@ -143,4 +142,80 @@ instance : Faithful toProfinite := {}
 example : toProfinite ⋙ profiniteToCompHaus = toCompHaus :=
   rfl
 
+/-- Every Stonean space is projective in `CompHaus` -/
+instance (X : Stonean) : Projective X.compHaus where
+  factors := by
+    intro B C φ f _
+    haveI : ExtremallyDisconnected X.compHaus.toTop := X.extrDisc
+    have hf : f.1.Surjective
+    · rwa [CompHaus.epi_iff_surjective] at *
+    obtain ⟨f', h⟩ := CompactT2.ExtremallyDisconnected.projective φ.continuous f.continuous hf
+    use ⟨f', h.left⟩
+    ext
+    exact congr_fun h.right _
+
 end Stonean
+
+namespace CompHaus
+
+/-- If `X` is compact Hausdorff, `presentation X` is an extremally disconnected space
+  equipped with an epimorphism down to `X`. It is a "constructive" witness to the
+  fact that `CompHaus` has enough projectives.  -/
+noncomputable
+def presentation (X : CompHaus) : Stonean where
+  compHaus := (projectivePresentation X).p
+  extrDisc := by
+    refine' CompactT2.Projective.extremallyDisconnected
+      (@fun Y Z _ _ _ _ _ _ f g hfcont hgcont hgsurj => _)
+    let g₁ : (CompHaus.of Y) ⟶ (CompHaus.of Z) := ⟨g, hgcont⟩
+    let f₁ : (projectivePresentation X).p ⟶ (CompHaus.of Z) := ⟨f, hfcont⟩
+    have hg₁ : Epi g₁ := (epi_iff_surjective _).2 hgsurj
+    refine' ⟨Projective.factorThru f₁ g₁, (Projective.factorThru f₁ g₁).2, funext (fun _ => _)⟩
+    change (Projective.factorThru f₁ g₁ ≫ g₁) _ = f _
+    rw [Projective.factorThru_comp]
+    rfl
+
+/-- The morphism from `presentation X` to `X`. -/
+noncomputable
+def presentationπ (X : CompHaus) : X.presentation.compHaus ⟶ X :=
```

Comment:
What do you think of chopping the name into an extra namespace part?
```suggestion
def presentation.π (X : CompHaus) : X.presentation.compHaus ⟶ X :=
```

Reply:
I agree that looks nicer

Reply:
When I change to `presentation.π`, I can't write `X.presentation.π` in the instance below anymore. It complains that "the environment does not contain `Stonean.π`, `presentation X` has type `Stonean`".

Reply:
No permutation of parentheses fixes the problem

Reply:
We could go with `presentation_π` and `presentation_epi_π` which I think is still nicer than the old names. 

Or we can write the instance below as 
```
instance presentation.epi_π (X : CompHaus) : Epi (π X)
```

What do you think?

Reply:
I've pushed your suggestion, but I'm happy with whatever

Reply:
Let's try it out like this. We can rename again later if we are unhappy.